superset-frontend/src/visualizations/TimeTable/TimeTable.jsx

```diff
@@ -27,13 +27,23 @@ import {
   MetricOption,
 } from '@superset-ui/chart-controls';
 import moment from 'moment';
+import sortNumericValues from 'src/utils/sortNumericValues';
 
 import FormattedNumber from './FormattedNumber';
 import SparklineCell from './SparklineCell';
 import './TimeTable.less';
 
 const ACCESSIBLE_COLOR_BOUNDS = ['#ca0020', '#0571b0'];
 
+const sortNumberWithMixedTypes = (rowA, rowB, columnId, descending) =>
+  sortNumericValues(
+    rowA.values[columnId].props['data-value'],
+    rowB.values[columnId].props['data-value'],
+    { descending, nanTreatment: 'asSmallest' },
+  ) *
+  // react-table sort function always expects -1 for smaller number
+  (descending ? -1 : 1);
```

Comment:
@graceguo-supercat @etr2460 I took a different approach and added the option to change how null/nans are treated. Please take a look